File: src/LC2026/LC053.java
package LC2026;

public class LC053 {

    public int maxSubArray(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int maxSum = nums[0];
        int curretMax = nums[0];
        for (int i = 1; i < nums.length; i++) {
            int cur = nums[i];
            curretMax = Math.max(cur, curretMax + cur);
            maxSum = Math.max(curretMax, maxSum);
        }
        return maxSum;
    }


}

File: srcKotlin/LCKotlin/LC053.kt
package LCKotlin

class LC053 {

    fun maxSubArray(nums: IntArray): Int {
        val dp: IntArray = IntArray(nums.size)
        var max = 0
        for (i in nums.indices) {
            if (i == 0) {
                dp[i] = nums[i]
                max = nums[i]
            } else {
                dp[i] = if (dp[i-1] > 0) {
                    dp[i-1] + nums[i]
                } else {
                    nums[i]
                }
                max = max.coerceAtLeast(dp[i])
            }
        }
        return max
    }
}